### PR TITLE
Use bazelisk as a way to manage bazel installation.

### PR DESCRIPTION
**Pull request by jasonwyatt**
This PR adds bazelisk binaries to `tools/` as well as a `.bazelversion` file to the project root, and will allow us to enforce that a particular version of bazel is always used. 

It also updates readmes/scripts where necessary to recommend `./bazelisk build` instead of `bazel build`.

This approach more closely follows the paradigm used by gradle, with `gradlew`, and should make it so we don't accidentally have the incorrect version of bazel installed for the build.

### CODE REVIEW

File: tools/src_kt_runtests.sh [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-echo "Testing java, javatests, and shells/android"
-bazel test \
-  --noshow_progress \
-  --noshow_loading_progress \
-  --test_output=errors \
-  --incompatible_depset_is_not_iterable=false \
-  //java/... //javatests/... //shells/android/javatests/...
-TEST_RESULT=$?
-
-exit $TEST_RESULT
```
Comment (SHeimlich) [inline tools/src_kt_runtests.sh]: Did you mean to delete this file? It's deletion feels unrelated to the rest of the PR.

Comment (jasonwyatt) [reply tools/src_kt_runtests.sh]: I deleted it because I felt updating it to use bazelisk was strange, given it was already broken :-/

Comment (SHeimlich) [reply tools/src_kt_runtests.sh]: Ahhh, gotcha.

**Comment by jasonwyatt** [comment]
This currently uses version 1.2 of Bazelisk, and upgrading should be as simple as downloading newly-released binaries from the https://github.com/bazelbuild/bazelisk page to replace the binaries we've got under `tools/`.

File: .bazelversion [hunk 2]
```diff
@@ -0,0 +1 @@
+1.2.0
```
Comment (csilvestrini) [inline .bazelversion]: Could you add a comment here to keep in sync with build_defs/bazel_version_check.bzl ?

Also modify that file to say to keep in sync with this one.

Comment (jasonwyatt) [reply .bazelversion]: Good idea, done
